.setup/install_submitty/get_globals.sh

```diff
@@ -0,0 +1,95 @@
+THIS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )/.."
+CONF_DIR="${THIS_DIR}/../../../config"
+
+VAGRANT=0
+if [ -d "${THIS_DIR}/../.vagrant" ]; then
+    VAGRANT=1
+fi
+
+UTM=0
+if [ -d "${THIS_DIR}/../.utm" ]; then
+    UTM=1
+fi
+
+CI=0
+if [ -f "${THIS_DIR}/../.github_actions_ci_flag" ]; then
+    CI=1
+fi
+
+SUBMITTY_REPOSITORY=$(jq -r '.submitty_repository' "${CONF_DIR}/submitty.json")
+SUBMITTY_INSTALL_DIR=$(jq -r '.submitty_install_dir' "${CONF_DIR}/submitty.json")
+WORKER=$([[ $(jq -r '.worker' "${CONF_DIR}/submitty.json") == "true" ]] && echo 1 || echo 0)
+
+# Because shellcheck is run with the python wrapper we need to disable the 'Not following' error
+# shellcheck disable=SC1091
+source "${THIS_DIR}/bin/versions.sh"
+
+if [ "${WORKER}" == 0 ]; then
+    ALL_DAEMONS=( submitty_websocket_server submitty_autograding_shipper submitty_autograding_worker submitty_daemon_jobs_handler )
+    RESTART_DAEMONS=( submitty_websocket_server submitty_daemon_jobs_handler )
+else
+    ALL_DAEMONS=( submitty_autograding_worker )
+    RESTART_DAEMONS=( )
+fi
+
+#############################################################
+# Re-Read other variables from submitty.json and submitty_users.json
+# (eventually will remove these from the /usr/local/submitty/.setup/INSTALL_SUBMITTY.sh script)
+
+SUBMITTY_DATA_DIR=$(jq -r '.submitty_data_dir' "${SUBMITTY_INSTALL_DIR}/config/submitty.json")
+# Worker does not need course builders so just use root
+if [ "${WORKER}" == 0 ]; then
+    COURSE_BUILDERS_GROUP=$(jq -r '.course_builders_group' "${SUBMITTY_INSTALL_DIR}/config/submitty_users.json")
+else
+    COURSE_BUILDERS_GROUP=root
+fi
+NUM_UNTRUSTED=$(jq -r '.num_untrusted' "${SUBMITTY_INSTALL_DIR}/config/submitty_users.json")
+FIRST_UNTRUSTED_UID=$(jq -r '.first_untrusted_uid' "${SUBMITTY_INSTALL_DIR}/config/submitty_users.json")
+FIRST_UNTRUSTED_GID=$(jq -r '.first_untrusted_gid' "${SUBMITTY_INSTALL_DIR}/config/submitty_users.json")
+DAEMON_USER=$(jq -r '.daemon_user' "${SUBMITTY_INSTALL_DIR}/config/submitty_users.json")
+DAEMON_GROUP=${DAEMON_USER}
+DAEMON_UID=$(jq -r '.daemon_uid' "${SUBMITTY_INSTALL_DIR}/config/submitty_users.json")
+DAEMON_GID=$(jq -r '.daemon_gid' "${SUBMITTY_INSTALL_DIR}/config/submitty_users.json")
+PHP_USER=$(jq -r '.php_user' "${SUBMITTY_INSTALL_DIR}/config/submitty_users.json")
+CGI_USER=$(jq -r '.cgi_user' "${SUBMITTY_INSTALL_DIR}/config/submitty_users.json")
+# Worker does not have daemon PHP so just use daemon group
+if [ "${WORKER}" == 0 ]; then
+    DAEMONPHP_GROUP=$(jq -r '.daemonphp_group' "${SUBMITTY_INSTALL_DIR}/config/submitty_users.json")
+else
+    DAEMONPHP_GROUP="${DAEMON_GROUP}"
+fi
+DAEMONCGI_GROUP=$(jq -r '.daemoncgi_group' "${SUBMITTY_INSTALL_DIR}/config/submitty_users.json")
+DAEMONPHPCGI_GROUP=$(jq -r '.daemonphpcgi_group' "${SUBMITTY_INSTALL_DIR}/config/submitty_users.json")
+SUPERVISOR_USER=$(jq -r '.supervisor_user' "${SUBMITTY_INSTALL_DIR}/config/submitty_users.json")
+
+export THIS_DIR
```

Comment:
I'm worried about the THIS_DIR variable.
It isn't very descriptive.
And since we are refactoring & adding new scripts & moving toward allowing these scripts to be run in smaller chunks (one at a time), and those scripts might be in different directories, it's going to be easy to make mistakes.

It should not be exported to the other scripts with this name.
Instead I think we should use a relative path from SUBMITTY_REPOSITORY or SUBMITTY_INSTALL_DIR...

Reply:
I agree it's poorly named; `THIS_DIR` was copied directly from the helper script. The problem with using a relative path from `SUBMITTY_REPOSITORY` or `SUBMITTY_INSTALL_DIR` is that those two variables are fetched from `${CONF_DIR}/submitty.json`, and to get `CONF_DIR` we need `THIS_DIR`. In the helper script, `THIS_DIR` is necessary to find `get_globals.sh`. With that said, we can remove the export since it is not used beyond this.

Reply:
Also, note that the `BASH_SOURCE` variable used to get `THIS_DIR` is relative to `get_globals.sh`, not the script sourcing it, so it's fine as a local variable.

Reply:
The .setup directory in the REPOSITORY?  
Or the .setup directory in the INSTALLATION DIRECTORY?
(yeah, the fact that I'm asking is part of the problem)

No need to make a global variable for either one, just relative off of whichever

Reply:
It's the `.setup` directory that the script is inside, which on the VM will always be the installation directory. But yes I see the problem.

Reply:
Nevermind, it's actually the one in the repository— I thought the install dir was where the scripts were run from; this is not the case. This is confusing.

Reply:
I've removed the export because the variable isn't actually used anywhere outside of `get_globals.sh`.